src/GitVersion.Core/Formatting/NumericFormatter.cs

```diff
@@ -20,10 +20,10 @@ public override bool TryFormat(object? value, string format, CultureInfo culture
             return true;
         }
 
-        // Hexadecimal formatting
-        if (format.StartsWith("X", StringComparison.OrdinalIgnoreCase) && int.TryParse(s, NumberStyles.Integer, cultureInfo, out var hex))
+        // Integer formatting with precision specifier
+        if ("BDX".Contains(char.ToUpperInvariant(format[0])) && int.TryParse(s, NumberStyles.Integer, cultureInfo, out var n))
```

Comment:
do you mind changing the variable from `n` to `numericValue`, and to make it consistent, please do the same for other similar variables in the method